### PR TITLE
#6Task. Elvin Zeinalabdziyeu

### PR DESCRIPTION
**Pull request by gurman-eez**
https://travis-ci.org/github/gurman-eez/js-assignments/builds/718156262

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -46,10 +53,15 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
-    throw new Error('Not implemented');
+	//  throw new Error('Not implemented');
+	for (let x = 1, i = 1; i <= n; i++) {
+		x *= i; 
+		if(i == n) return x
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator === is always preferable
sometimes ''==' may cause an error
(check at console :
""0"" == 0
and ""0"" === 0)"

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -63,10 +75,17 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+	//  throw new Error('Not implemented');
+	for (let z = 0,i = n1; i <= n2; i++) {
+		z += i;
+		if (i == n2) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -82,7 +101,8 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+	//  throw new Error('Not implemented');
+	return ((a < b + c) && (b < a + c) && (c < b + a)) ? true : false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js:107]: we can return exactly ((a < b + c) && (b < a + c) && (c < b + a)) with the same result
Please, see https://javascript.info/logical-operators

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -150,7 +184,23 @@ function doRectanglesOverlap(rect1, rect2) {
...
+			yp = point.y;
+
+	// calculate the distance with the Pythagorean theorem
+	const distance = Math.sqrt((xp - xc)**2 + (yp - yc)**2);
+
+	// match the distance with radius
+	if (distance < radius) {
+		return true 
+	}
+	return false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js:205]: we can return exactly distance < radius, which will be converted to true or false
Please, see https://javascript.info/logical-operators

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -166,7 +216,14 @@ function isInsideCircle(circle, point) {
  *   'entente' => null
  */
 function findFirstSingleChar(str) {
-    throw new Error('Not implemented');
+	//  throw new Error('Not implemented');
+	for (let i = 0; i < str.length; i++) {
+		let c = str.charAt(i);
+		if (str.indexOf(c) == i && str.indexOf(c, i + 1) == -1) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -373,10 +536,22 @@ function toNaryString(num, n) {
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
  */
 function getCommonDirectoryPath(pathes) {
-    throw new Error('Not implemented');
+	//  throw new Error('Not implemented');
+	let res = "";
+	let el = pathes.map(e => e.split("/"))
+	console.log(el)
+	for (let i = 0; i < el[0].length; i++) {
+		if (el.every(e => e[i] == el[0][i])) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -431,7 +612,30 @@ function getMatrixProduct(m1, m2) {
  *
  */
 function evaluateTicTacToePosition(position) {
-    throw new Error('Not implemented');
+	//  throw new Error('Not implemented');
+	const check = (a, b, c) => {
+		return (a === b && b === c && a != undefined) ? true : false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
""0"" != 0
and ""0"" !== 0)"

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by vramaniuk** [comment]
> https://travis-ci.org/github/gurman-eez/js-assignments

Please, privide the direct link to your task success travis build
And resolve conflicting files(readme.md)

File: task/06-conditions-n-loops-tasks.js (hunk 1 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js]: fixed

File: task/06-conditions-n-loops-tasks.js (hunk 2 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js]: fixed

File: task/06-conditions-n-loops-tasks.js (hunk 3 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js:107]: ok thanks

File: task/06-conditions-n-loops-tasks.js (hunk 4 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js:205]: thanks for advice!

File: task/06-conditions-n-loops-tasks.js (hunk 7 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js]: fixed

File: task/06-conditions-n-loops-tasks.js (hunk 6 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js]: fixed

File: task/06-conditions-n-loops-tasks.js (hunk 5 above)
Comment (gurman-eez) [reply task/06-conditions-n-loops-tasks.js]: fixed

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**